botocore/utils.py

```diff
@@ -48,6 +48,20 @@
 RETRYABLE_HTTP_ERRORS = (requests.Timeout, requests.ConnectionError)
 S3_ACCELERATE_WHITELIST = ['dualstack']
 
+CONTROLS = {
+    '\n': '\\n',
+    '\r': '\\r',
+    '\t': '\\t',
+}
+# Combines all CONTROLS keys into a big or regular expression
+ESCAPE_CONTROLS_RE = re.compile('|'.join(map(re.escape, CONTROLS)))
+# Based on the match get the appropriate replacement from CONTROLS
+CONTROLS_MATCH_HANDLER = lambda match: CONTROLS[match.group(0)]
```

Comment:
Just as a heads up, we usually try to avoid using `lambdas` and defer to defining a function. While for this case it is not bad, the main reason is that following tracebacks in lambdas can be hard.